src/app/components/tfrs/tfrs.component.ts

```diff
@@ -1,13 +1,124 @@
-import { Component, OnInit } from '@angular/core';
+
 import { ActivatedRoute } from '@angular/router';
+import { Component, OnInit, AfterViewInit, ViewChild } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { tfrService } from 'src/app/service/tfrs/tfr.service';
+import {MatTableDataSource} from '@angular/material/table';
+import {MatPaginator} from '@angular/material/paginator';
+import { MatSort,Sort } from '@angular/material/sort';
+import { LiveAnnouncer } from '@angular/cdk/a11y';
+
+
+
+export interface PeriodicElement {
+  id: String;
+  name: String;
+  vendor_id: String;
+  start_date: String;
+  end_date:String;
+  status:String;
+  version:String;
+  vendor_specific:String;
+  is_deleted:String;
+  created_by:String;
+  modified_by:String;
+  created_at:String;
+  modified_at:String
+}
 
 @Component({
   selector: 'app-tfrs',
   templateUrl: './tfrs.component.html',
   styleUrls: ['./tfrs.component.scss'],
 })
-export class TfrsComponent implements OnInit {
-  constructor(private route: ActivatedRoute) {}
+export class TfrsComponent implements OnInit, AfterViewInit {
+
+  displayedColumns: string[] = ['name', 'start_date', 'end_date','status' ,'link'];
+  ELEMENT_DATA:any=[];
+  projectList:MatTableDataSource<PeriodicElement>=new MatTableDataSource(this.ELEMENT_DATA)
+  selectedVendorName:any;
+  vendors:any
+  statusList:any=['DRAFT','INPROGRESS','ARCHIVED','DELIVERED']
```

Comment:
Please move this in constant files as other can also use same variable for example Rishali have graph widget where we are using this